src/mmrelay/matrix_utils.py

```
@@ -1106,34 +1256,63 @@ async def connect_matrix(passed_config=None):
+                    try:
+                        if credentials is not None:
+                            credentials["device_id"] = e2ee_device_id
+                            save_credentials(credentials)
+                            logger.info(
+                                "Updated credentials.json with discovered device_id"
+                            )
+                    except (IOError, OSError) as e:
+                        logger.warning(f"Failed to persist discovered device_id: {e}")
+                else:
+                    logger.warning("whoami response did not contain device_id")
             except Exception as e:
```

Ruff (0.14.5)

Do not catch blind exception: `Exception`

(BLE001)